packages/trader/src/Modules/Contract/Containers/contract-replay.tsx

```diff
@@ -0,0 +1,384 @@
+    isSmartTraderContract,
+    urlFor,
+} from '@deriv/shared';
+import { localize } from '@deriv/translations';
+import { useFeatureFlags } from '@deriv/hooks';
+import ChartLoader from 'App/Components/Elements/chart-loader';
+import ContractDrawer from 'App/Components/Elements/ContractDrawer';
+import UnsupportedContractModal from 'App/Components/Elements/Modals/UnsupportedContractModal';
+import SmartChartSwitcher from '../../Trading/Containers/smart-chart-switcher.jsx';
+import { ChartBottomWidgets, ChartTopWidgets, DigitsWidget, InfoBoxWidget } from './contract-replay-widget.jsx';
+import ChartMarker from 'Modules/SmartChart/Components/Markers/marker.jsx';
+import DelayedAccuBarriersMarker from 'Modules/SmartChart/Components/Markers/delayed-accu-barriers-marker';
+import allMarkers from 'Modules/SmartChart/Components/all-markers.jsx';
+import ChartMarkerBeta from 'Modules/SmartChartBeta/Components/Markers/marker.jsx';
+import { observer, useStore } from '@deriv/stores';
+import { useTraderStore } from 'Stores/useTraderStores';
+
+type TLocation = Location & { state: { from_table_row: boolean } | Record<string, never> };
+
+const ContractReplay = observer(({ contract_id }: { contract_id: number }) => {
+    const { common, contract_replay, ui } = useStore();
+    const [swipe_index, setSwipeIndex] = React.useState(0);
+    const { contract_store } = contract_replay;
+    const {
+        is_market_closed,
+        is_sell_requested,
+        onClickCancel,
+        onClickSell,
+        onMount,
+        onUnmount,
+        indicative_status,
+        is_chart_loading,
+        is_forward_starting,
+    } = contract_replay;
+    const { contract_info, contract_update, contract_update_history, is_digit_contract } = contract_store;
+    const { routeBackInApp } = common;
+    const { is_dark_mode_on: is_dark_theme, notification_messages_ui: NotificationMessages, toggleHistoryTab } = ui;
+    const trade_type_feature_flag =
+        contract_info.shortcode &&
+        getContractTypeFeatureFlag(contract_info.contract_type ?? '', isHighLow(contract_info));
+    const is_trade_type_disabled =
+        useFeatureFlags()[`is_${trade_type_feature_flag}_enabled` as keyof ReturnType<typeof useFeatureFlags>] ===
+        false;
+    const [is_visible, setIsVisible] = React.useState(false);
+    const history = useHistory();
+
+    React.useEffect(() => {
+        const url_array = /[^/]*$/.exec(location.pathname);
+        const url_contract_id = url_array ? +url_array[0] : undefined;
+        onMount(contract_id || url_contract_id);
+        setIsVisible(true);
+
+        return () => {
+            setIsVisible(false);
+            onUnmount();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [contract_id, location, onMount, onUnmount]);
+
+    const onClickClose = React.useCallback(() => {
+        setIsVisible(false);
+        const is_from_table_row = !isEmptyObject((location as TLocation).state)
```

Comment:
By default `location` has `Location` type, but there is no such field as state in it. Created expanded `TLocation` (line 44)